.github/docker/windows/entrypoint.ps1

```diff
@@ -0,0 +1,60 @@
+#
+# Entrypoint that installs given python version and runs tests from commit.
+#
+# $python has to be a full Python version like 3.7.5, supported by chocolatey.
+# $repo is the repository like 'nerdvegas/rez'
+# $commit is the revision to test
+param( $python, $repo, $commit )
+
+# Stop on errors; .exe has to be checked manually
+Set-StrictMode -Version latest
+$ErrorActionPreference = "Stop"
+
+echo "Testing $repo at $commit"
+
+# Fixes encoding issue on Windows 10 local docker run.
+#
+${ENV:PYTHONIOENCODING} = "UTF-8"
+
+# Only 2.x respects the install dir. 3.x will install to C:\Python3x
+choco install python --yes --version=$python --installargs='TargetDir=""C:\python""'
```

Comment:
I feel like we should do the install into the Dockerfile. The goal of docker is to have a reproducible build, we we won't have with the install in the entrypoint.

Could we install all version in their respective folder in the dockerfile and append the one we test into $PATH here?

So in the Dockerfile:

```
RUN choco install python --yes --version=2.7.11 --installargs='TargetDir=""C:\python2.7""' \
    && choco install python --yes --version=3.6.8 --installargs='TargetDir=""C:\python3.6""' \
    && choco install python --yes --version=3.7.5 --installargs='TargetDir=""C:\python3.7""'
```

And here:

```
${ENV:PATH} += ";C:\python"
${ENV:PATH} += ( $python.Split(".")[0..1] -Join '' )
```
(note my noobeness in Windows here on the last line, I don't know if we can reference a variable inside a string...).

This proposition would speed up our builds and would make the tests really reproducible. What do you think?

Reply:
Good point. I guess the reason I did as I did is so we would need to maintain the images each time we want to change python Versions. The install does not add that much to the test time - but I was originally considering three images (since the cost is small due to layering in docker). In retrospect installing all three versions would work for me, too.

Btw. I was not able to control the target dir for 3.x python, but the default were reasonable so we can make it work by just installing to c:\python27

Reply:
I am leaning towards putting it into the Dockerfile. Making the builds as reproducible and deterministic as possible would beat slightly lower maintenance efforts. Looking at the recent python release dates that looks bearable to me.